src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DataBaseHandler.java

```diff
@@ -0,0 +1,233 @@
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is the database handler class. at initiation it checks if the name given exist as a db file,
+ * if not it gets created together with required tables. At initiation a file name is given making
+ * it possible to store a file per project depending how the classes that uses this handler
+ * implements this function.
+ *
+ */
+public class DataBaseHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataBaseHandler.class);
+    private static final String EVENT_ID_KEY = "eventId";
+    private final String databaseFile;
+
+    /**
+     * Constructor that takes plugin directory path as String and plugin name as String.
+     *
+     * @param pluginDir
+     * @param filename
+     * @throws ConnectException
+     */
+    public DataBaseHandler(final String pluginDir, final String filename) throws ConnectException {
+        Path filePath = Paths.get(pluginDir, filename);
+        this.databaseFile = String.format("jdbc:sqlite:%s", filePath);
+        createNewDatabase();
+        createTables();
+    }
+
+    /**
+     * Constructor that takes plugin directory path as File and plugin name as String.
+     *
+     * @param pluginDir
+     * @param filename
+     * @throws ConnectException
+     */
+    public DataBaseHandler(final File pluginDir, final String filename) throws ConnectException {
+        this(pluginDir.toString(), filename);
+    }
+
+    /**
+     * This sets up an enum for Tables including the key name used as primary key for that table
+     *
+     */
+    public enum Table {
+        SCS_TABLE("branch"), SCC_TABLE("changeId");
+
+        private final String keyName;
+
+        Table(String keyValue) {
+            this.keyName = keyValue;
+        }
+
+        public String getKeyName() {
+            return this.keyName;
+        }
+    }
+
+    /**
+     * This function returns an event id if exists for a specific table depending on the keyValue
+     *
+     * @param table
+     * @param keyValue
+     * @return eventId
+     * @throws ConnectException
+     */
+    public String getEventID(final Table table, final String keyValue) throws ConnectException {
+        String eventID = "";
+
+        String sqlSelectStatement = String.format("SELECT * FROM %s WHERE %s=?", table, table.keyName);
+        try (Connection connection = connect();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlSelectStatement)) {
+
+            preparedStatement.setString(1, keyValue);
+            eventID = executeQuery(preparedStatement);
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to fetch values from database: {}\n{}", e.getMessage(), e);
```

Comment:
You should maybe rethrow as another error and not return an empty eventid. It's then up to the one using this functionality to handle the error.

Reply:
Exception is thrown